### PR TITLE
Added error handling for XGB_CONFIGURED event

### PR DESCRIPTION
**Pull request by nvidianz**
Fixes FLARE-2152.

### Description

1. Ignore TenSEAL errors if not in horizontal secure training mode
2. Added error handling after XGB_CONFIGURED event
3. Fixed an error in PR #2771 where a wrong header is used. 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by yanchengnv — CHANGES_REQUESTED** [review]
See my comments about event firing.

File: nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py [hunk 1]
```diff
@@ -402,22 +404,24 @@ def _process_after_all_gather_v_horizontal(self, fl_ctx: FLContext):
         fl_ctx.set_prop(key=Constant.PARAM_KEY_RCV_BUF, value=result, private=True, sticky=False)
 
     def handle_event(self, event_type: str, fl_ctx: FLContext):
+        global tenseal_error
         if event_type == Constant.EVENT_XGB_JOB_CONFIGURED:
             training_mode = fl_ctx.get_prop(Constant.PARAM_KEY_TRAINING_MODE)
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py:409]: It is set inside the xgb_adaptor.py line 192

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def initialize(self, fl_ctx: FLContext):
         self._workspace = fl_ctx.get_prop(FLContextKey.WORKSPACE_OBJECT)
         run_number = fl_ctx.get_prop(FLContextKey.CURRENT_RUN)
         self._run_dir = self._workspace.get_run_dir(run_number)
-        self.engine = engine
+        self.engine = fl_ctx.get_engine()
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py:52]: All are fixed and still checking.

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py [hunk 3]
```diff
@@ -212,10 +212,10 @@ def _check_duplicate_seq(self, op: str, rank: int, seq: int):
         with self._lock:
             event = self._pending_req.get((rank, seq), None)
         if event:
-            self.log_info(fl_ctx, f"Duplicate seq {op=} {rank=} {seq=}, wait till original req is done")
+            self.logger.info(f"Duplicate seq {op=} {rank=} {seq=}, wait till original req is done")
             event.wait(DUPLICATE_REQ_MAX_HOLD_TIME)
             time.sleep(1)  # To ensure the first request is returned first
-            self.log_info(fl_ctx, f"Duplicate seq {op=} {rank=} {seq=} returned with empty buffer")
+            self.logger.info(f"Duplicate seq {op=} {rank=} {seq=} returned with empty buffer")
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py:218]: Fixed.

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py:52]: Sorry I miss these places in the PR, I am testing the integration test with all the old codes and existing example.

I am also adding and testing E2E new example based on V2

I will make sure all xgboost example run good before we build

File: nvflare/app_opt/xgboost/histogram_based_v2/executor.py [hunk 4]
```diff
@@ -124,7 +124,12 @@ def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort
                 fl_ctx,
             )
             engine.fire_event(Constant.EVENT_XGB_JOB_CONFIGURED, fl_ctx)
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/app_opt/xgboost/histogram_based_v2/executor.py:126]: Good idea. Changed.

File: nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py [hunk 5]
```diff
@@ -402,22 +404,24 @@ def _process_after_all_gather_v_horizontal(self, fl_ctx: FLContext):
...
             if training_mode in {"vertical_secure", "vs"} and ipcl_imported:
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py:410]: Done.

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py [hunk 6]
```diff
@@ -212,10 +212,10 @@ def _check_duplicate_seq(self, op: str, rank: int, seq: int):
         with self._lock:
             event = self._pending_req.get((rank, seq), None)
         if event:
-            self.log_info(fl_ctx, f"Duplicate seq {op=} {rank=} {seq=}, wait till original req is done")
+            self.logger.info(f"Duplicate seq {op=} {rank=} {seq=}, wait till original req is done")
```
[the remark this answers is not in the document]
Comment (nvidianz) [reply nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py:215]: All fixed.

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM

**Pull request merged**